### ISSUE TITLE
 Add collections templates

### ISSUE THREAD

**Issue by falzm** [facette/facette#170]
As for graph templates (#112), implement "collection templates" allowing collections to be created dynamically using templates.

**Comment by zecrazytux** [comment]
hey,

has any work been done on this subject yet ?
thx !

**Comment by falzm** [comment]
Hi @zecrazytux 

No progress on this matter for now, we'll try to free some time this weekend.

**Comment by falzm** [comment]
Backend implementation has been drafted on branch [collection-templates](https://github.com/facette/facette/tree/collection-templates).

To @vbatoufflet for UI implementation:

Example of collection template:

```json
{
    "id": "a3269a61-930d-49f6-6bf3-ee7701e1bb01",
    "name": "test_template",
    "title": "{{ .source }}",
    "description": "Collection of graphs for source {{ .source }}",
    "template": true,
    "entries": [
        {
            "id": "84d6ef76-83a5-43a9-41b1-96d9600a4eb4",
            "options": {
                "constants": [
                    null
                ],
                "enabled": true,
                "percentiles": [
                    null
                ],
                "range": null,
                "title": null
            }
        }
    ]
}
```

Example of template-based collection:

```json
{
    "id": "2f72b448-f498-432a-5ffe-5fad12b3f45f",
    "name": "host1.example.net",
    "title": "host1.example.net",
    "description": "Collection of graphs for source host1.example.net",
    "template": false,
    "attributes": { "source": "host1.example.net" },
    "entries": [
        {
            "id": "84d6ef76-83a5-43a9-41b1-96d9600a4eb4",
            "attributes": { "port": 8081 }
        }
    ]
}
```

Note: the UI-side plot fetching code is responsible for template collection/entries attributes merging, should query the backend such as:

```shell
$ curl -X POST \
  -H 'Content-Type: application/json' \
  -d '
{
  "range": "-1h",
  "graph": {
    "link": "84d6ef76-83a5-43a9-41b1-96d9600a4eb4",
    "attributes": {
      "source": "host1.example.net",
      "port": 8081
    }
  }
}
' \
  localhost:12003/api/v1/plots
```

**Comment by zecrazytux** [comment]
Hey,

I noticed your work on this subject, thanks. It is still missing the UI bits to be usable, right ?

**Comment by falzm** [comment]
Implemented in branch [0.4-alpha](https://github.com/facette/facette/tree/0.4-alpha).
